### PR TITLE
Remove positive transform for anisotropic lengthscales

### LINKED ISSUE #1639: Anisotropic lengthscales
Currently, by default, all stationary kernels impose a positive transform on the lengthscales Parameter. For the case of AnisotropicStationary kernels, this reduces the viable parameter space by a factor of 2^(D-1), where D is the number of input dimensions. So it is acceptable in the case of a 1D problem, but in higher dimensions we would miss out. For example, for the Cosine kernel there is a significant difference between lengthscales of [+1, +1], and [+1, -1], they correspond to waves propagating in perpendicular directions.

### PR DESCRIPTION
**Pull request by frgsimpson**
Resolves #1639

### CODE REVIEW

File: gpflow/kernels/stationaries.py [hunk 1]
```diff
@@ -115,6 +115,12 @@ class AnisotropicStationary(Stationary):
     input dimension.
     """
 
+    def __init__(self, *args, **kwargs):
```
Comment (vdutor) [inline gpflow/kernels/stationaries.py:118]: Best to copy the base class `__init__` parameters for better docstrings and autocomplete.

File: gpflow/kernels/stationaries.py [hunk 2]
```diff
@@ -115,6 +115,12 @@ class AnisotropicStationary(Stationary):
...
+        super().__init__(*args, **kwargs)
+
+        if tf.size(self.lengthscales) > 1:
```
Comment (vdutor) [inline gpflow/kernels/stationaries.py:121]: I would prefer to do it like we do it in `ard`:
```suggestion
  if self.lengthscales.shape.ndims > 1
```

**Review by vdutor — APPROVED** [review]
LGTM

File: gpflow/kernels/stationaries.py (hunk 2 above)
Comment (frgsimpson) [reply gpflow/kernels/stationaries.py]: Nice, I've updated to just use self.ard itself

File: gpflow/kernels/stationaries.py (hunk 1 above)
Comment (frgsimpson) [reply gpflow/kernels/stationaries.py]: Done!

**Comment by st--** [comment]
Superseded by #1843

**Pull request closed without merging**